internal/workload/manager_test.go

```diff
@@ -92,6 +101,23 @@ var _ = Describe("Manager", func() {
 
 			// then
 			Expect(err).NotTo(HaveOccurred())
+			for i := 0; i < 10; i++ {
+				wkName := fmt.Sprintf("test%d", i)
+				pod := getPodFor(datadir, wkName)
+				Expect(pod.Name).To(BeEquivalentTo(wkName))
+
+				Expect(pod.Spec.Containers).To(HaveLen(1))
+				Expect(pod.Spec.Containers[0].Env).To(HaveLen(1))
```

Comment:
Just a note, on loops in case of any fail, will be hard to know what pod failed, maybe adding some kind of message on the assert: 
```
Expect(pod.Spec.Containers[0].Env[0]).To(BeEquivalentTo(v1.EnvVar{Name: "DEVICE_ID", Value: deviceId}), fmt.Sprintf("Failing on pod %s", wkName))
```

Reply:
Done